core/server/master/src/test/java/alluxio/underfs/UfsStatusCacheTest.java

```diff
@@ -446,6 +448,91 @@ public void testFetchSingleStatusThrowsException() throws Exception {
     Mockito.verify(mUfs, times(1)).getStatus(any(String.class));
   }
 
+  @Test
+  public void testMetricCacheSizeTotal() throws Exception{
+
+    AlluxioURI path0 = new AlluxioURI("/abc/0");
+    UfsStatus stat0 = Mockito.mock(UfsStatus.class);
+    when(stat0.getName()).thenReturn("0");
+
+    AlluxioURI path1 = new AlluxioURI("/abc/1");
+    UfsStatus stat1 = Mockito.mock(UfsStatus.class);
+    when(stat1.getName()).thenReturn("1");
+
+    //a wrong path
+    AlluxioURI path2 = new AlluxioURI("/abc/0");
+    UfsStatus stat2 = Mockito.mock(UfsStatus.class);
+    when(stat2.getName()).thenReturn("1");
+
+    mCache.addStatus(path0, stat0);
+    mCache.addStatus(path1, stat1);
+    assertEquals(2,DefaultFileSystemMaster.Metrics.UFS_STATUS_CACHE_SIZE_TOTAL.getCount());
+    //repeatedly add
+    mCache.addStatus(path1, stat1);
+    assertEquals(2,DefaultFileSystemMaster.Metrics.UFS_STATUS_CACHE_SIZE_TOTAL.getCount());
+    try{
+      mCache.addStatus(path2, stat2);
+    }catch (Exception ignored){}
+    assertEquals(2,DefaultFileSystemMaster.Metrics.UFS_STATUS_CACHE_SIZE_TOTAL.getCount());
+
+    mCache.remove(path0);
+    assertEquals(1,DefaultFileSystemMaster.Metrics.UFS_STATUS_CACHE_SIZE_TOTAL.getCount());
+
+    mCache.remove(path1);
+    assertEquals(0,DefaultFileSystemMaster.Metrics.UFS_STATUS_CACHE_SIZE_TOTAL.getCount());
+
+    //remove repeatedly
+    mCache.remove(path1);
+    assertEquals(0,DefaultFileSystemMaster.Metrics.UFS_STATUS_CACHE_SIZE_TOTAL.getCount());
+
+    mCache.remove(path2);
+    assertEquals(0,DefaultFileSystemMaster.Metrics.UFS_STATUS_CACHE_SIZE_TOTAL.getCount());
+  }
+
+  @Test
+  public void testMetricChildrenSizeTotal() throws Exception{
+    AlluxioURI path = new AlluxioURI("/abc");
+    UfsStatus stat = Mockito.mock(UfsStatus.class);
+    when(stat.getName()).thenReturn("abc");
+    mCache.addStatus(path, stat);
+
+    //a children list ['0','1','2']
+    ArrayList<UfsStatus> statusList0=new ArrayList<>();
+    for (int i=0;i<3;i++){
+      UfsStatus statChild = Mockito.mock(UfsStatus.class);
+      when(statChild.getName()).thenReturn(String.valueOf(i));
```

Comment:
even if you need to mock something, try to extract a method to avoid duplication